src/instantsearch/instantsearch.ts

```diff
@@ -146,6 +147,7 @@ export type SearchForFacetValuesResponse = {
 };
 
 export type SearchClient = {
+  addAlgoliaAgent: (agent: string) => void;
```

Comment:
It doesn't necessarily exist if it's a custom search client.

```suggestion
  addAlgoliaAgent?: (agent: string) => void;
```